pages/api/xAPI/statements/index.ts

```diff
@@ -149,96 +145,89 @@ async function addSingleStatement(
   req: NextRequest,
   res: NextResponse,
   next: Next
-): void {
-
-}
+): void {}
 
 async function addMultipleStatements(
-  req: NextRequest,
-  res: NextResponse,
-): void {
-
-}
-
-async function handlePOST(
   req: NextRequest,
   res: NextResponse
-): void {
-  const client = new MongoClient(uri, options);
-  if(!req.body)
-  {
+): void {}
+
+async function handlePOST(req: NextRequest, res: NextResponse): void {
+  if (!req.body) {
     res.status(200).end();
   }
   try {
-    await client.connect();
-    helpers.sanitizeBody(req, res, (err)=> {throw err;});
-    let body = req.body.data
+    await dbClient.connect();
+    helpers.sanitizeBody(req, res, (err) => {
+      throw err;
+    });
+    let body = req.body.data;
     body.forEach((statement) => {
-      statement['stored'] = new Date(Date.now()).toISOString();
-      if (!statement.timestamp){
-        statement['timestamp'] = statement.stored;
+      statement["stored"] = new Date(Date.now()).toISOString();
+      if (!statement.timestamp) {
+        statement["timestamp"] = statement.stored;
       }
     });
-    body = helpers.createStatementID(body, res, (err) => {throw err;})
+    body = helpers.createStatementID(body, res, (err) => {
+      throw err;
+    });
     req.body.data = body;
-    const ids = body.map(statement => statement.id);
+    const ids = body.map((statement) => statement.id);
     let result;
-    if(body.length > 1) {
-      result = await client
+    if (body.length > 1) {
+      result = await dbClient
         .db("xAPI")
         .collection("statements")
         .insertMany(body);
     } else {
-      result = await client
+      result = await dbClient
         .db("xAPI")
         .collection("statements")
         .insertOne(body[0]);
     }
-    if(!result.acknowledged || !result.insertedId) {
+    if (result.insertedCount == 0) {
       throw "failed to record statement";
     }
     res.status(200).send(ids);
-  } catch(err) {
-    if(err.name === 'MongoServerError') {
-      if(err.code === 11000) {
-        return runDupeCheck(req, res, err);
+  } catch (err) {
+    if (err.name === "MongoServerError") {
+      if (err.code === 11000) {
+        return await runDupeCheck(req, res, err);
       }
     }
     throw err;
   } finally {
-    client.close();
+    await dbClient.close();
```

Comment:
Added `await` to prevent the method from returning while executing `.close()`.

In other words, having `await dbClient.close();` ensures `.close()` completes before handlePost() returns.